src/main/java/Duke.java

```diff
@@ -1,10 +1,48 @@
+import command.Command;
+import data.*;
+
+/**
+ * The Duke iP Program.
+ */
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Starts the program.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
```

Comment:
Hey Im not sure if its common to use isExit, would it be clearer if you use isDone / isCompleted instead?